rewrite-java-core/src/test/java/BinaryDataTest.java

```diff
@@ -0,0 +1,92 @@
+import com.azure.recipes.v2recipes.TypeReferenceRecipe;
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+public class BinaryDataTest implements RewriteTest {
+    /**
+     * BinaryDataTest tests the recipe that changes
+     * com.azure.core.util.BinaryData to io.clientcore.core.util.binarydata.BinaryData.
+     * This recipe also tests to ensure that the TypeReference is correctly changed
+     * when used in conjunction with BinaryData.
+     * @author Ali Soltanian Fard Jahromi
+     */
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new TypeReferenceRecipe());
+    }
+
+    /* Test to make sure BinaryData type and import is changed */
+    @Test
+    public void testBinaryDataTypeAndImportChanged() {
+        @Language("java") String before = "import com.azure.core.util.BinaryData;";
+        before += "\npublic class Testing {";
+        before += "\n  public Testing(){";
+        before += "\n    com.azure.core.util.BinaryData b = BinaryData.fromObject(null);";
```

Comment:
We have an import above. So, do we need to use fully qualified name here?

Reply:
Fixed in commit [492daeb](https://github.com/Azure/azsdk-java-rewrite-recipes/pull/14/commits/492daeb48360470829cc54218911b9b3e1758ace)